### PR TITLE
Initial draft for viewport to improve OS rendering.

### PR DESCRIPTION
**Pull request by thmasjs**
# Changes

*Describe the changes made, including any changes to tests, dependencies or docs. Link to any relevant issues below.*

- Closes #XXXX *(replace XXXX with GitHub issue number)*
- ...

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

File: packages/vis-core/src/Components/Sidebar/Selectors/SelectorSection.test.jsx [hunk 1]
```diff
@@ -161,6 +161,41 @@ let props = {
...
+        <MapContext.Provider value={{ state: mockState, dispatch }}>
+          <FilterContext.Provider value={mockFilterContext}>
+            <SelectorSection {...props} />
+          </FilterContext.Provider>
+        </MapContext.Provider>
+      </AppContext.Provider>
+    );
+
+    expect(screen.queryByText("Viewport")).not.toBeInTheDocument();
+    expect(screen.getByText("Hazard")).toBeInTheDocument();
```
Comment (brereh) [inline packages/vis-core/src/Components/Sidebar/Selectors/SelectorSection.test.jsx]: Does this test pass? Seems to be related to CVT directly.

File: packages/vis-core/src/Components/MapLayout/Map.jsx [hunk 2]
```diff
@@ -1124,6 +1129,32 @@ const Map = (props) => {
     }
   }, [isMapReady]);
 
+  // Keep `state.currentZoom` in sync with the actual map zoom.
```
Comment (brereh) [inline packages/vis-core/src/Components/MapLayout/Map.jsx]: This should be in as an atomic 'fix' commit.

**Comment by thmasjs** [comment]
Sorry, large confusion with my initial implementation. 

Have fixed and this should be worked with: https://github.com/Transport-for-the-North/TAME-Web-API/pull/221 and https://github.com/Transport-for-the-North/CVT-Visualisation-Framework/pull/7

File: packages/vis-core/docs/Viewport Documentation.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+In your layer config:
+
+```js
+{
+  name: "OS Road Network",
+  type: "tile",
+  source: "api",
+  path: "/api/vectortiles/cvt_os_roads/{z}/{x}/{y}",
+
+  appendViewportParamsToTiles: true,
```
Comment (brereh) [inline packages/vis-core/docs/Viewport Documentation.md]: By definition, the vector tiles will be bounded to the viewport, so the related functionality is not needed.

File: packages/vis-core/src/contexts/MapContext.jsx [hunk 4]
```diff
@@ -139,20 +139,29 @@ export const MapProvider = ({ children }) => {
...
           case 'mapFeatureSelect':
           case 'mapFeatureSelectWithControls':
           case 'mapFeatureSelectAndPan':
+          case 'mapViewport':
             filters.push(filterWithId);
             break;
 
           default:
-            switch (filter.values.source) {
+            // Some filter types don't define `values` (or configs may be incomplete).
```
Comment (brereh) [inline packages/vis-core/src/contexts/MapContext.jsx]: Is this the only change beyond minor refactoring here?
